DOC: change wording on new contributor path

tacaswell opened the pull request.

rcomer reviewed (commented):
Wording looks good but the `--` renders as is.  Is that intentional?

tacaswell commented:
I was hoping this would become emdashes when rendered.  We appear to have explicitly turned them off: https://github.com/matplotlib/matplotlib/blob/80bc1ac0ac72bcb78a071d83e820de9656ea58e3/doc/conf.py#L618-L619 via aae1b6e14ea0b37479390c78fdba3438b454a250

I fixed the two I added and one other in this file.

tacaswell commented:
Ah, the context is https://github.com/matplotlib/matplotlib/blob/80bc1ac0ac72bcb78a071d83e820de9656ea58e3/doc/conf.py#L618-L619 which was messing up quotation marks.
